### PR TITLE
Add meter mod

### PR DESCRIPTION
**Pull request by GraysonWu**
Add meterMod and meterBand.

### CODE REVIEW

**Comment by GraysonWu** [comment]
Hi @jojimt, I have this PR that needs your review. It's required by Project Antrea. Thank you so much.

File: openflow13/meter.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	n := 0
+	m.MeterBandHeader.UnmarshalBinary(data[n:])
+	n += int(m.MeterBandHeader.Len())
+
+	return nil
+}
+
+type MeterBandDSCP struct {
+	MeterBandHeader       /* Type: OFPMBT13_DSCP_REMARK. */
+	PrecLevel       uint8 /* Number of drop precedence level to add. */
```
Comment (antoninbas) [inline openflow13/meter.go:107]: it's the opposite I think

> amount by which the drop precedence of the packet should be reduced if the band is exceeded

Comment (GraysonWu) [reply openflow13/meter.go:107]: I'm trying to follow the definition here: https://github.com/openvswitch/ovs/blob/master/include/openflow/openflow-1.3.h#L185

File: openflow13/meter.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+	return &MeterBandHeader{
+		Type:      OFPMBT13_DROP,
+		Length:    16,
+		Rate:      100,
+		BurstSize: 200,
+	}
+}
+
+func (m *MeterBandHeader) Len() (n uint16) {
+	return 12
```
Comment (jojimt) [inline openflow13/meter.go]: Please avoid hard coding.

File: openflow13/meter.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+func (m *MeterBandHeader) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, 12)
```
Comment (jojimt) [inline openflow13/meter.go]: Here too.

File: openflow13/meter.go [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+func (m *MeterBandDrop) Len() (n uint16) {
+	return m.MeterBandHeader.Len() + 4
+}
+
+func (m *MeterBandDrop) MarshalBinary() (data []byte, err error) {
+	data, err = m.MeterBandHeader.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	bytes := make([]byte, 4)
```
Comment (jojimt) [inline openflow13/meter.go]: If you use sizeof in the marshal function, you shouldn't have to do this append here.

File: openflow13/meter.go [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+type MeterBandDSCP struct {
+	MeterBandHeader       /* Type: OFPMBT13_DSCP_REMARK. */
+	PrecLevel       uint8 /* Number of drop precedence level to add. */
+	pad             [3]uint8
+}
+
+func (m *MeterBandDSCP) Len() (n uint16) {
+	return m.MeterBandHeader.Len() + 4
```
Comment (jojimt) [inline openflow13/meter.go]: why not just use sizeof, especially when you defined a pad?

File: openflow13/meter.go [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+	return nil
+}
+
+type MeterBandExperimenter struct {
+	MeterBandHeader
+	Experimenter uint32 /* Experimenter ID which takes the same form as in struct ofp_experimenter_header. */
+}
+
+func (m *MeterBandExperimenter) Len() (n uint16) {
+	return m.MeterBandHeader.Len() + 4
```
Comment (jojimt) [inline openflow13/meter.go]: please avoid hard coding

File: openflow13/meter.go [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+	}
+
+	return
+}
+
+func (m *MeterMod) MarshalBinary() (data []byte, err error) {
+	m.Header.Length = m.Len()
+	data, err = m.Header.MarshalBinary()
+
+	bytes := make([]byte, 8)
```
Comment (jojimt) [inline openflow13/meter.go]: Please use sizeof, if you can.

**Comment by antoninbas** [comment]
Thanks for the review @jojimt! @GraysonWu is on vacation right now but will address your comments next week when he comes back to work.

**Comment by GraysonWu** [comment]
Hi @jojimt, thanks for the review. I just pushed a version with your comments addressed. 
Because the length of all `MeterBand*` are fixed which are 16. Instead of using sizeof, I define a const of theirs length to avoid append. You can take a look to see if this way works for you.

**Pull request merged**